app/models/user.rb

```diff
@@ -6,8 +6,19 @@ class User < ActiveRecord::Base
 
   has_many :lists
 
+  before_create do |doc|
```

Comment:
A model spec would be a good place to test that the api key is being generated when a new user is created.